### PR TITLE
Build OSX Conda Nightly

### PR DESCRIPTION
**Pull request by Skylion007**
## Motivation and Context
* Build Nightly OSX Binaries
@mathfac Putting this here so I can get some help testing and deploying the CI command.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,6 +6,43 @@ gpu: &gpu
   environment:
     FPS_THRESHOLD: 900
 commands:
+  build_osx_conda_pkg:
+    description: "Build OSX Conda-Package"
+    parameters:
```
Comment (mathfac) [inline .circleci/config.yml:11]: It's better to have same parameter signature like [the linux building job](https://github.com/facebookresearch/habitat-sim/blob/f216a1732a2d33747c56df9a4d07485ba70cda3a/.circleci/config.yml#L617-L626):

```
    parameters:
      AIHABITAT_CONDA_CHN:
        type: string
        default: "aihabitat"
      AIHABITAT_CONDA_CHN_PWD_VAR:
        type: string
        default: "AIHABITAT_CONDA_PWD"
      NIGHTLY_FLAG:
        type: string
        default: ""
```
 Maybe, add one more parameter `upload` for cases when no upload required.

File: .circleci/config.yml [hunk 2]
```diff
@@ -6,6 +6,43 @@ gpu: &gpu
...
+      - run:
+          name: Conda OSX build
+          command: |
+            export PATH=$HOME/miniconda/bin:$PATH
+            source ~/.bash_profile
+            conda activate
+            conda install -y anaconda-client ccache cmake git ninja conda-build pip
+            echo $(which -a python)
+            pip install gitpython
+            <<^ parameters.test>> anaconda login --username $AIHABITAT_CONDA_CHN --password $AIHABITAT_CONDA_CHN_PWD
```
Comment (mathfac) [inline .circleci/config.yml:39]: Here channel is known and nonsensitive information, while password is taken from environment variable and parameter is used only for the variable name:

```suggestion
            anaconda login --username << parameters.AIHABITAT_CONDA_CHN >>  --password $<< parameters.AIHABITAT_CONDA_CHN_PWD_VAR >>
```

**Review by mathfac — CHANGES_REQUESTED** [review]
Ideally the one job should serve 3 use-cases:
- CI test build with limited python versions support without upload that will run on every commit
- CI nightly that will build nightly package from master and upload it
- Version release that will build the packages and upload it to main Habitat conda channel.

**Comment by mathfac** [comment]
The error you have right now is not related to the anaconda, but pre installed dependencies:
`cp: cannot stat '/usr/include/EGL': No such file or directory`

**Comment by erikwijmans** [comment]
> @erikwijmans The conda nightly builds are currently being built, after they are done I will add them as a dependency on build_conda_binaries so they are run if the Linux conda build succeeds.

I guess the dependency is since the linux one is responsible for removing old binaries? If so, we can factor that out in the future, but no need to do that now.

**Comment by Skylion007** [comment]
Exactly

> 
> 
> > @erikwijmans The conda nightly builds are currently being built, after they are done I will add them as a dependency on build_conda_binaries so they are run if the Linux conda build succeeds.
> 
> I guess the dependency is since the linux one is responsible for removing old binaries? If so, we can factor that out in the future, but no need to do that now.

Exactly

**Review by erikwijmans — APPROVED** [review]
Ship it!

**Pull request merged**